enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+  teams, and covered by those components’ unit and CI testing.
+* Decouple the definition of single node developer cluster
+  configuration from the SNC tooling and CRC installer, allowing it to
+  potentially be reused in various other contexts. For example, single
+  node developer clusters could be deployed directly using the
+  standard OpenShift installer or some new specialized installer, or
+  VM images could be created by means other than the current SNC
+  tooling. Any such use cases would require separate enhancement PRs.
+* Ensure that single node developer OpenShift clusters are as
+  conformant as possible - that they pass most Kubernetes and
+  OpenShift conformance end-to-end tests. Fully document any tests
+  that must be skipped due to differences from a traditional
+  deployment.
+
+### Non-Goals
+
+* Single node clusters supportable for production or critical task
+  workloads.  The cluster profile described in this enhancement
+  proposal is not intended to support any form of production cluster
+  deployment. Single node clusters for non-developer use cases, such
+  as for edge computing, would require distinct cluster profiles with
+  independent selection and configuration of components specific to
+  those use cases.
+* A new CRC installer. Although it should somewhat simplify the SNC
+  tooling used for CRC, this enhancement proposal does not change the
+  basic approach to installing CRC or producing a VM image for the CRC
+  installer. By making the definition of the content and configuration
+  of the components in a single node developer cluster more orthogonal
+  to the tooling that deploys the cluster, this enhancement could
+  enable distinct future enhancements to improve the build and/or
+  install process used by CRC.
+* A new OpenShift installer. This enhancement proposal does not add a
+  new OpenShift installer or involve changes, beyond the ability to
+  specify the cluster profile, to the standard OpenShift installer
+  currently used to build CRC installation images. Evolution of the
+  existing installer, SNC tooling and other CRC components is
+  expected.
+* Functional changes for OpenShift. This enhancement proposal does not
+  result in any functional changes to clusters that do not utilize the
+  single-node-developer cluster profile.
+
+## Proposal
+
+The single-node-developer cluster profile will be implemented in two
+phases. The first phase introduces the new cluster profile, and
+enables some simplification of the SNC image build process by
+utilizing the profile to control which components are included in the
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
```

Comment:
which manifests? pods? deployments? crds?

Reply:
It can be any manifest contained in the release image (/release-manifests directory in the image). It can be a PrometheusRule, a Service, a Deployment, a CRD, etc.

Reply:
We could start by adding the annotation to absolutely every manifest that is normally included in the default (self-managed-high-availability) profile. But we want to save some time and effort by avoiding adding the annotation to manifests that are definitely not needed for CRC, such as those removed by the SNC image build tooling or otherwise found to be unneeded, as described below.